File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/script/ScriptCacheTest.java
package net.bestia.zoneserver.script;

import org.junit.Before;
import org.junit.Test;
import static org.mockito.Mockito.*;

import java.io.File;

public class ScriptCacheTest {

	private final static String VALID_SCRIPT = "valid.js";
	private ScriptCache cache;

	private ScriptCompiler compiler;
	private ScriptFileResolver resolver;
	private File validFile = new File("/valid/script/valid.js");

	@Before
	public void setup() {

		compiler = mock(ScriptCompiler.class);
		resolver = mock(ScriptFileResolver.class);

		when(resolver.getScriptFile(eq(VALID_SCRIPT), any(ScriptType.class)))
				.thenReturn(validFile);

		cache = new ScriptCache(compiler, resolver);
	}

	@Test(expected = NullPointerException.class)
	public void addFolder_nullFolder_throws() {
		cache.addFolder(null, ScriptType.ATTACK);
	}

	@Test
	public void addFolder_validFolder_compilesContent() {

	}

	@Test(expected = NullPointerException.class)
	public void getScript_nullName_throws() {
		cache.getScript(ScriptType.ATTACK, null);
	}

	@Test
	public void getScript_validNameAndType_compiles() {
		cache.getScript(ScriptType.ATTACK, VALID_SCRIPT);
		
		verify(compiler).compileScript(validFile);
	}

}

File: bestia-zoneserver/src/test/kotlin/net/bestia/zoneserver/script/ScriptCacheTest.kt
package net.bestia.zoneserver.script

import com.nhaarman.mockitokotlin2.any
import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.verify
import com.nhaarman.mockitokotlin2.whenever
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner
import javax.script.CompiledScript

@RunWith(MockitoJUnitRunner::class)
class ScriptCacheTest {

  @Mock
  lateinit var compiler: ScriptCompiler

  @Mock
  lateinit var scriptFileResolver: ScriptFileResolver

  private val requestedScriptName = "ibims1script.js"

  private lateinit var cache: ScriptCache

  @Before
  fun setup() {
    val compiledScript = mock<CompiledScript>()
    whenever(scriptFileResolver.getScriptInputStream(any())).thenReturn(mock())
    whenever(compiler.compileScript(any())).thenReturn(compiledScript)

    cache = ScriptCache(compiler, scriptFileResolver)
  }

  @Test
  fun `getScript() returns the compiled script`() {
    val cachedScript = cache.getScript(requestedScriptName)
    verify(scriptFileResolver).getScriptInputStream(requestedScriptName)
    val cachedScript2 = cache.getScript(requestedScriptName)
    Assert.assertEquals("Script was not cached.", cachedScript, cachedScript2)
  }
}